src/schema/definition.rs

```diff
@@ -10,25 +18,45 @@ use value::Key;
 #[derive(Debug)]
 pub struct Definition {
   /// Types defined in the database.
-  types: Vec<Type>
+  types: BTreeMap<Key, Type>
 }
 
 impl Definition {
   /// Creates a new empty instance of `Definition`.
   pub fn new() -> Self {
     Definition {
-      types: Vec::new()
+      types: BTreeMap::new()
     }
   }
 
   /// Add a new type to the `Definition`.
-  pub fn add_type(&mut self, type_: Type) {
-    self.types.push(type_);
+  pub fn add_type<K>(&mut self, name: K, type_: Type) where K: Into<Key> {
+    self.types.insert(name.into(), type_);
   }
 
   /// Gets type of a certain name.
-  pub fn find_type<N>(&self, name: &N) -> Option<&Type> where N: PartialEq<Key> {
-    self.types.iter().find(|type_| name.eq(type_.name()))
+  pub fn get_type(&self, name: &Key) -> Option<&Type> {
```

Comment:
A variation of that worked. Don't know if this builds with the parameters we'd expect. We'll see when we get there…